### ISSUE TITLE
--reprocess flag on `pachctl edit` does not reprocess datums

### ISSUE THREAD

**Issue by mattusifer** [pachyderm/pachyderm#4200]
**What happened?**:
I ran `pachctl edit pipeline --reprocess`, which triggered another job in that pipeline in which all datums were skipped (not reprocessed).

**What you expected to happen?**:
I expected all datums in that pipeline to be reprocessed, but instead they were skipped

**How to reproduce it (as minimally and precisely as possible)?**:
Run `pachctl edit pipeline --reprocess` on a pipeline, make a very minimal change.

**Anything else we need to know?**:

**Environment?**:
- Kubernetes version (use `kubectl version`): 1.15.2
- Pachyderm CLI and pachd server version (use `pachctl version`): 1.9.7
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s): aws (EKS)
- OS (e.g. from /etc/os-release):
- Others:

**Comment by JoeyZwicker** [comment]
I'm not seeing a reprocess as a flag on pachctl edit. I'm not sure we actually even need edit given we have `update pipeline`.

Usage:
  pachctl edit [command]

Available Commands:
  pipeline    Edit the manifest for a pipeline in your text editor.

Global Flags:
      --no-color   Turn off colors.
  -v, --verbose    Output verbose logs

Use "pachctl edit [command] --help" for more information about a command.

**Comment by mattusifer** [comment]
I'm seeing it as a flag on the pipeline subcommand:
```
pachctl edit pipeline -h                                                                                                                                                                                     ✱ add-internal-fdqn-output-from-gateway
Edit the manifest for a pipeline in your text editor.

Usage:
  pachctl edit pipeline <pipeline> [flags]

Flags:
      --editor string   Editor to use for modifying the manifest.
      --reprocess       If true, reprocess datums that were already processed by previous version of the pipeline.

Global Flags:
      --no-color   Turn off colors.
  -v, --verbose    Output verbose logs
```

I've found it useful because update pipeline requires a JSON file, or otherwise requires all information about the pipeline. If i just want to quickly trigger a reprocess I can use `edit` and make a minor change to the layout.

That would be a good feature if it doesn't already exist - the ability to trigger a reprocess without updating the pipeline. Does that exist?

**Comment by JoeyZwicker** [comment]
`update pipeline` works great even if you just pass it the same manifest that's already running, but I understand the value of what you're asking with `edit pipeline`.

**Comment by mattusifer** [comment]
My problem with `update pipeline` is that my manifest is dynamically generated during CI (we actually use the `python_pachyderm` library for deployment), so I don't generally have a manifest on hand that I can pass into it.

**Comment by JoeyZwicker** [comment]
FYI, you can use `pachctl extract pipeline <pipeline> | pachctl update pipeline`.

**Comment by mattusifer** [comment]
That is good to know! That's exactly what I need. Thanks.

**Comment by pappasilenus** [comment]
@JoeyZwicker I confirmed that the `--reprocess` flag exists in 1.9.7.

**Comment by jdoliner** [comment]
This is a real bug the issue is that while reprocess is being set in the request, the salt for the pipeline is also being set which is overriding the reprocessing behavior.

**Comment by jdoliner** [comment]
Fixed by #4265.
